完整pipeline的基本搭建，文件上传pipeline测试完毕

xxxlwb opened the pull request:
llm调用那一块工具内目前是固定选择实例化的LLMInterface("opai","gpt-4o")，为本地测试先写成这样
embedding那一块是mock的function

xxxlwb committed:
完善数据库连接管理，添加对数据库 URI 为 None 的检查，并修复关闭连接的方法

xxxlwb committed:
优化api目录文件处理和内存检索逻辑，修改参数错误

xxxlwb committed:
添加 python-multipart、tiktoken 和 magic-pdf 依赖

xxxlwb committed:
更新 README 文档，调整 API 启动说明；修改提取 PDF 的逻辑，优化文本提取和错误处理；更新知识图谱构建器的参数类型注解。

HUANG-429 committed:
Update tools-relavant files

Fix functionality in tools package; Add test script to verify the correct flow of orchestrator; Edit knowledge_graph.models to support tools; Add helpful markdown files

HJY-429 committed:
Delete test/test_requirements.txt

HJY-429 committed:
Update README_TEST.md

HUANG-429 committed:
Fix metadata handlings

Separate file_meta and request_meta; Links should be included separately in files part instead of request_metadata for batch files; Add pipeline_parameters.md showing how params are being passed; Adjust test package

xxxlwb committed:
feat: Add httpx dependency and update OpenAI API key handling

- Added `httpx[socks]` dependency to `pyproject.toml`.
- Updated `OPENAI_API_KEY` in `base.py` to include a default value for testing.
- Commented out the `Base.metadata.drop_all(bind=engine)` line in `db.py` to prevent accidental data loss.
- Created a new Jupyter notebook `test_llm.ipynb` for testing LLM interactions.
- Enhanced `ToolResult` serialization in `base.py` to handle nested `ToolResult` objects.
- Added logging for source data retrieval and version hash calculation in `blueprint_generation_tool.py`.
- Updated `document_etl_tool.py` to log content extraction and RawDataSource creation.
- Improved logging in `graph_build_tool.py` for better traceability during graph processing.
- Refactored `PipelineOrchestrator` to use a global LLM client and improved tool registration.
- Introduced `route_wrapper.py` to handle file uploads and JSON requests for the tools system.

HUANG-429 committed:
Update MemoryGraphBuildTool to support system running.

Add logging info for orchestrator and api_integration for error handling;
Update MemoryGraphBuildTool and memory_system to support system running;
Update api/ingest.py for correct json data response;
Adjust content_vec param in KnowledgeBlock table for default embedding model;
Translate Chinese comments to English in route_wrapper.py;
Fix syntax error in cognitive_map.py

IANTHEREAL commented on api/README.md:
```diff
@@ -7,14 +7,19 @@ This module provides a REST API for the Narrative Knowledge system, enabling doc
 
 ### 2. Start the API Server
-
+#### You can start the API using either method:
```
```suggestion
You can start the API using either method:
```

IANTHEREAL commented on api/README.md:
```diff
@@ -7,14 +7,19 @@ This module provides a REST API for the Narrative Knowledge system, enabling doc
 ### 2. Start the API Server
-
+#### You can start the API using either method:
+##### Method 1: Run the main module directly
```
```suggestion
#### Method 1: Run the main module directly
```

IANTHEREAL commented on api/README.md:
````diff
@@ -7,14 +7,19 @@ This module provides a REST API for the Narrative Knowledge system, enabling doc
 ```bash
 poetry run python -m api.main
 ```
+##### Method 2: Use uvicorn (recommended for production)
````
```suggestion
#### Method 2: Use uvicorn (recommended for production)
```

IANTHEREAL commented on api/README.md:
```diff
@@ -222,4 +227,258 @@ The API uses standardized error responses for all endpoints:
+        "timestamp": "2025-08-01T22:55:24.658312+00:00"
+      }
+    },
+    "pipeline": ["MemoryGraphBuildTool"],
```
why it is inconsistent with the request?

HJY-429 replied:
For memory processing, we currently use default pipeline (MemoryGraphBuild), regardless of `process_strategy`.

IANTHEREAL commented on api/README.md:
```diff
@@ -222,4 +227,258 @@ The API uses standardized error responses for all endpoints:
+    },
+    "BlueprintGenerationTool": {
+      "max_chunk_size": 4000,
+      "overlap": 200
```
What are the functions of these two parameters of BlueprintGenerationTool?

IANTHEREAL commented on api/ingest.py:
```diff
@@ -239,11 +239,106 @@ async def save_data(
+
+from tools.route_wrapper import ToolsRouteWrapper
+
+# 在模块级别创建包装器实例
```
Use english for codes and comments

HJY-429 committed:
Update README.md

HJY-429 committed:
Update models.py

Use default embedding model.

IANTHEREAL commented on api/ingest.py:
```diff
@@ -239,11 +239,106 @@ async def save_data(
+        
+        result = tools_wrapper.process_json_request(
+            input_data=body.get("input"),
+            metadata=body.get("metadata", {}),
```
use the argument `metadata` directly

IANTHEREAL commented on knowledge_graph/congnitive_map.py:
```diff
@@ -294,7 +294,13 @@ def _generate_document_cognitive_map(self, topic_name: str, document: Dict) -> D
 
         try:
-            response = self.llm_client.generate(cognitive_map_prompt)
+            logger.info(
```
ditto

IANTHEREAL commented on knowledge_graph/graph.py:
```diff
@@ -200,7 +204,13 @@ def generate_analysis_blueprint(
             )
-            response = self.llm_client.generate(blueprint_prompt, max_tokens=8192)
+            # 为跑通流程暂时这样写
+            from llm.factory import LLMInterface
```
ditto

IANTHEREAL commented on knowledge_graph/graph.py:
```diff
@@ -246,12 +256,25 @@ def generate_analysis_blueprint(
+                # 更新已有记录（例如更新内容或状态）
+                blueprint.processing_items = blueprint_items
+                blueprint.processing_instructions = processing_instructions
+                blueprint.status = "outdated"
```
but it is updated, not outdated

IANTHEREAL commented on llm/providers/gemini.py:
```diff
@@ -25,38 +25,46 @@ def __init__(self, model: str, **kwargs):
-                        if part.text:
-                            yield part.text
+                    for candidate in resp.candidates:
+                        for part in candidate.content.parts:
```
Had you find a bug here?

IANTHEREAL commented on memory_system.py:
```diff
@@ -152,8 +152,8 @@ def __init__(
         """
-        self.llm_client = llm_client
-        self.embedding_func = embedding_func or get_text_embedding
+        self.llm_client = LLMInterface("openai", model="gpt-4o") if llm_client is None else llm_client
```
As suggested above, this is not a good idea.

IANTHEREAL commented on knowledge_graph/models.py:
```diff
@@ -284,15 +351,31 @@ class AnalysisBlueprint(Base):
+    )
+    # Track which source data contributed to this blueprint
+    contributing_source_data_ids = Column(JSON, nullable=False)  # List of source_data IDs
+    source_data_version_hash = Column(String(64), nullable=True)  # Hash of source data versions
```
remove it if we don't use it

IANTHEREAL commented on setting/base.py:
```diff
@@ -18,8 +18,8 @@
 # DB settings
-DATABASE_URI = os.environ.get("DATABASE_URI")
-SESSION_POOL_SIZE: int = os.environ.get("SESSION_POOL_SIZE", 40)
+DATABASE_URI = os.environ.get("DATABASE_URI","mysql+pymysql://2v2WvLxTSjY5adC.root:2rA6B5Zc98PgMqNH@gateway01.ap-southeast-1.prod.alicloud.tidbcloud.com:4000/test?ssl_ca=/etc/ssl/cert.pem&ssl_verify_cert=true&ssl_verify_identity=true")
```
lol, now our database is leaked, please keep it in the configuration file or environment variable and do not commit

IANTHEREAL commented on setting/db.py:
```diff
@@ -23,7 +21,7 @@
 )
 
-Base = declarative_base()
+# Base = declarative_base()
```
why comment it, and no bug after commenting it?

IANTHEREAL commented on tools/test_integration.py:
```diff
@@ -0,0 +1,118 @@
+"""
```
put it under test/pipline/

xxxlwb commented on etl/extract.py:
```diff
@@ -61,12 +68,15 @@ def convert_pdf_to_markdown_using_pymupdf(pdf_path):
-            text = page.get_text()  # get plain text encoded as UTF-8
+        for page_num in range(total_pages):
+            page = doc.load_page(page_num)
+            text_page = page.get_textpage()  # get TextPage object
```
My editor vscode cannot find the get_text() function. The official documentation says that the get_text() function needs to pass in parameters, but the static syntax parser pylance keeps reporting an error saying that it cannot find this function. I looked up the official pymupdf 1.26.3 library and found another way to write it, which did not report an error.

HUANG-429 committed:
Update api to support batch files uploading

Update api/ingest and toolswrapper to support batch files uploading; Remove test scripts to test package; Update setting/base to hide database_url; Fix other reference errors

xxxlwb committed:
- Cleaned up whitespace and formatting  for better code clarity.
- Standardized comments and documentation across various methods for better understanding.

xxxlwb committed:
Merge branch 'fix/chinese_comments_to_english'

HUANG-429 committed:
Reorganize files uploading methods

Files uploading has two methods to handle; Add log feedback for memory processing with explicit pipelines

xxxlwb committed:
重构知识图谱相关模型，更新数据源处理逻辑，确保数据库连接URI不为空

xxxlwb committed:
更新文件上传API，增强参数验证，确保文件与链接数量匹配，并优化元数据处理逻辑

HUANG-429 committed:
Modify memory uploading process to be asynchronous

Edit json processing in /save POST; Store messages in api/memory; Fetch and process source_id in PersonalMemorySystem;Add asynchronous methods to track tasks status

HUANG-429 committed:
Fix some problems in memory asynchronous process

Add BackgroundTask table to track task status; Modify links input to be optional

xxxlwb committed:
修改多文件上传的蓝图构建逻辑

xxxlwb committed:
fix readme

HUANG-429 committed:
Add asynchronous files uploading

Modify files uploading to be asynchronous; Add new table in knowledge_graph/models to track background task status; Fix error handling in repeated inputs

xxxlwb committed:
fix restful api

xxxlwb committed:
fix restful api

xxxlwb committed:
delete ipynb

HJY-429 committed:
Update knowledge.py

Adjust _generate_build_id() to include topic_name as part of the generation of raw_data_source (PRIMARY KEY) id.

xxxlwb committed:
delete link index uniqueness constraint in SourceData model

xxxlwb committed:
Optimize knowledge graph processing logic, simplify the return value structure, and fix the handling of duplicate files across different topics.

HUANG-429 committed:
Fix force_regenerate parameter passing

xxxlwb committed:
Fixed data query in document ETL tool to ensure existing source data is filtered based on topic name

xxxlwb committed:
daemon_pipeline

xxxlwb committed:
fix blueprint generation strategy , fix daemon,fix document_etl_tool

xxxlwb committed:
fix default pipeline

HUANG-429 committed:
Fix param passing in files daemon part; Add memory process in pipeline daemon

Modify daemon input args to accept both files and memory; Since chat messages are stored in RawDataSource, clean them after successful processing to save storage space; Fix relavant errors in data storage and reading; Improve orchestrator logics

HJY-429 committed:
Update README.md

Adjust indentation in pipeline daemon.

xxxlwb committed:
fix retrieval,daemon,background_task

xxxlwb committed:
Fixed build ID filtering conditions in RawDataSource queries and added file upload logs; updated embedded functions and environment variable configurations.

HUANG-429 committed:
Update tools-relavant settings

HUANG-429 committed:
Add uploading scripts for toc_files

Add uploading scripts for toc_files, with one small size for testing and another whole size; Fix daemon to group a maximum of 5 files each time; TODO: trace back RDS data when backgroundTask is failed

HJY-429 committed:
Update daemon.py

Fix syntax error

HUANG-429 committed:
Fix reference error in daemon

HUANG-429 committed:
Add KnowledgeBuildTool to construct knowledge blocks

Modify orchestrator and daemon to accept this new tool; Edit setting and llm to use ollama model

HUANG-429 committed:
Fix KnowledgeBuild process logic

HUANG-429 committed:
Fix KnowledgeBuild process logic
